src/main/java/fi/hsl/transitdata/pulsarpubtransconnect/PubtransTableHandler.java

```diff
@@ -133,77 +142,48 @@ public void setLastModifiedTimeStamp(long ts) {
-        String operatingDay;
-
-        JourneyInfo(Map<String, String> map) {
-            direction = map.get(TransitdataProperties.KEY_DIRECTION);
-            routeName = map.get(TransitdataProperties.KEY_ROUTE_NAME);
-            startTime = map.get(TransitdataProperties.KEY_START_TIME);
-            operatingDay = map.get(TransitdataProperties.KEY_OPERATING_DAY);
-            if (!isValid()) {
-                //Let's print more info for debugging purposes:
-                log.warn("JourneyInfo is missing some fields. Content: " + this.toString());
+    protected Optional<PubtransTableProtos.DOITripInfo> getTripInfo(long dvjId, long jppId) {
+        try {
+            String stopIdKey = TransitdataProperties.REDIS_PREFIX_JPP + Long.toString(jppId);
+            Optional<String> maybeStopId = Optional.ofNullable(jedis.get(stopIdKey));
+
+            String tripInfoKey = TransitdataProperties.REDIS_PREFIX_DVJ + Long.toString(dvjId);
+            Optional<Map<String, String>> maybeTripInfoMap = Optional.ofNullable(jedis.hgetAll(tripInfoKey));
+
+            if (maybeStopId.isPresent() && maybeTripInfoMap.isPresent()) {
+                PubtransTableProtos.DOITripInfo.Builder builder = PubtransTableProtos.DOITripInfo.newBuilder();
+                builder.setStopId(maybeStopId.get());
+                maybeTripInfoMap.ifPresent(map -> {
+                    if (map.containsKey(TransitdataProperties.KEY_DIRECTION))
+                        builder.setDirectionId(Integer.parseInt(map.get(TransitdataProperties.KEY_DIRECTION)));
+                    if (map.containsKey(TransitdataProperties.KEY_ROUTE_NAME))
+                        builder.setRouteId(map.get(TransitdataProperties.KEY_ROUTE_NAME));
+                    if (map.containsKey(TransitdataProperties.KEY_START_TIME))
+                        builder.setStartTime(map.get(TransitdataProperties.KEY_START_TIME));
+                    if (map.containsKey(TransitdataProperties.KEY_OPERATING_DAY))
+                        builder.setOperatingDay(map.get(TransitdataProperties.KEY_OPERATING_DAY));
+                });
+                builder.setDvjId(dvjId);
+                return Optional.of(builder.build());
+            }
+            else {
+                log.error("Failed to get data from Redis for dvjId {}, jppId {}", dvjId, jppId);
+                return Optional.empty();
             }
         }
-
-        boolean isValid() {
-            return direction != null && routeName != null &&
-                   startTime != null && operatingDay != null;
-        }
-
-        @Override
-        public String toString() {
-            return new StringBuilder()
-                    .append("direction: ").append(direction)
-                    .append(" routeName: ").append(routeName)
-                    .append(" startTime: ").append(startTime)
-                    .append(" operatingDay: ").append(operatingDay)
-                    .toString();
+        catch (Exception e) {
```

Comment:
So this is thrown if some tripinfo fields are missing?

Reply:
yes, builder.build throws a runtime exception if the data is not valid or required fields are not set. I don't really like this so I have been explicitly adding throw-clauses to some factory-methods that do this.